app/controllers/samples_reports_controller.rb

```diff
@@ -0,0 +1,133 @@
+class SamplesReportsController < ApplicationController
+  include SamplesReportsHelper
+
+  helper_method :boxes_data
+  helper_method :available_institutions
+
+  def index
+    @can_create = has_access?(@navigation_context.institution, CREATE_INSTITUTION_SAMPLES_REPORT)
+    @can_delete = has_access?(SamplesReport, DELETE_SAMPLES_REPORT)
+
+    @samples_reports = SamplesReport.where(institution: @navigation_context.institution)
+    @samples_reports = check_access(@samples_reports, READ_SAMPLES_REPORT).order('samples_reports.created_at DESC')
+  
+    # Filter by search params
+
+    @samples_reports = @samples_reports.partial_name(params[:name])
+    @samples_reports = @samples_reports.partial_sample_uuid(params[:sample_uuid])
+    @samples_reports = @samples_reports.partial_box_uuid(params[:box_uuid])
+    @samples_reports = @samples_reports.partial_batch_number(params[:batch_number])
+
+    #paginate samples report
+    @samples_reports = perform_pagination(@samples_reports)
+  end
+  
+  def new
+    @samples_report = SamplesReport.new({
+      institution: @navigation_context.institution,
+      site: @navigation_context.site
+    })
+
+    @boxes = []
+    
+    prepare_for_institution_and_authorize(@samples_report, CREATE_INSTITUTION_SAMPLES_REPORT)
+  end
+
+  def create
+    @institution = @navigation_context.institution
+    return unless authorize_resource(@institution, CREATE_INSTITUTION_SAMPLES_REPORT)
+
+    @samples_report = SamplesReport.new()
+    @samples_report.institution = @institution
+    @samples_report.site = @navigation_context.site
+    @samples_report.name = params[:samples_report][:name]
+    
+    samples_report_samples = []
+    @boxes = [] # For form reloading in case of validation error
+    if params[:samples_report][:boxes_attributes]
+      @boxes_p = params[:samples_report][:boxes_attributes]
+      @boxes_p.each do |box_param|
+        box_id = @boxes_p[box_param][:box_id]
+        box = Box.find_by_id(box_id)
```

Comment:
The actual shape of the params is the following:

```
{
    samples_report: 
    {
        name => "1", 
        boxes_attributes => 
        { 
            "15672403" => 
            {
                "box_id" => 48
            }
        }
    }
}
```

This is because we were reusing the `BoxSelector` component used on `TransferPackages` here. (see the conversation between Gus/Johannes/me above about why it is like this. I was maintaining the `params` format used for transfers, but now that you mention this, I don't see any reason to don't change it for the case of the reports to your suggestion of

```
{
  samples_report: {
    box_ids: [123, 456]
  }
}

```

I'll go with that and also fix the security issue :) 

Thanks!! <3